src/test/WormholeJoin.t.sol

```diff
@@ -614,6 +614,91 @@ contract WormholeJoinTest is DSTest {
         assertEq(join.totalDebt(), 250_000 * RAD);
     }
 
+    function testRegisterAndWithdrawPayingOperatorFee() public {
+        WormholeGUID memory guid = WormholeGUID({
+            sourceDomain: "l2network",
+            targetDomain: "ethereum",
+            receiver: addressToBytes32(address(123)),
+            operator: addressToBytes32(address(this)),
+            amount: 250_000 ether,
+            nonce: 5,
+            timestamp: uint48(block.timestamp)
+        });
+        assertEq(vat.dai(address(this)), 0);
+        uint256 daiSent = join.requestMint(guid, 0, 250 ether);
+        assertEq(vat.dai(address(this)), 250 * RAD);
+        assertEq(dai.balanceOf(address(123)), 249_750 ether);
+        assertEq(_pending(guid), 0);
+        assertEq(_ink(), 250_000 ether);
+        assertEq(_art(), 250_000 ether);
+        assertEq(daiSent, 249_750 * WAD);
+    }
+
+    function testFailOperatorFeeTooHigh() public {
+        WormholeGUID memory guid = WormholeGUID({
+            sourceDomain: "l2network",
+            targetDomain: "ethereum",
+            receiver: addressToBytes32(address(123)),
+            operator: addressToBytes32(address(this)),
+            amount: 250_000 ether,
+            nonce: 5,
+            timestamp: uint48(block.timestamp)
+        });
+        join.requestMint(guid, 0, 250_001 ether);   // Slightly over the amount
+    }
+
+    function testRegisterAndWithdrawPartialPayingOperatorFee() public {
+        WormholeGUID memory guid = WormholeGUID({
+            sourceDomain: "l2network",
+            targetDomain: "ethereum",
+            receiver: addressToBytes32(address(123)),
+            operator: addressToBytes32(address(this)),
+            amount: 250_000 ether,
+            nonce: 5,
+            timestamp: uint48(block.timestamp)
+        });
+
+        join.file("line", "l2network", 200_000 ether);
+        join.requestMint(guid, 0, 200 ether);
+
+        assertEq(vat.dai(address(this)), 200 * RAD);
+        assertEq(dai.balanceOf(address(123)), 199_800 ether);
+        assertTrue(_blessed(guid));
+        assertEq(_pending(guid), 50_000 ether);
+        assertEq(_ink(), 200_000 ether);
+        assertEq(_art(), 200_000 ether);
+
+        join.file("line", "l2network", 250_000 ether);
+        join.mintPending(guid, 0, 5 ether);
+
+        assertEq(vat.dai(address(this)), 205 * RAD);
+        assertEq(dai.balanceOf(address(123)), 249_795 ether);
+        assertEq(_pending(guid), 0);
+        assertEq(_ink(), 250_000 ether);
+        assertEq(_art(), 250_000 ether);
+    }
+
+    function testRegisterAndWithdrawPayingTwoFees() public {
```

Comment:
i could think of one more test: fail if operatorFee is bigger than `amtToTake - fee`. This will overflow.

Reply:
done